test/trento/activity_logging/activity_logger_test.exs

```diff
@@ -196,4 +196,46 @@ defmodule Trento.ActivityLog.ActivityLoggerTest do
       end
     end
   end
+
+  test "domain event activity logging" do
+    heartbeat_succeeded_event = build(:heartbeat_succeded)
+    heartbeat_failed_event = build(:heartbeat_failed)
+    host_registered_event = build(:host_registered_event)
+    host_checks_health_changed_event = build(:host_checks_health_changed)
+    host_checks_selected_event = build(:host_checks_selected)
+
+    software_updates_discovery_requested_event =
+      build(:software_updates_discovery_requested_event)
+
+    events = [
+      {host_registered_event, "host_registered"},
+      {heartbeat_succeeded_event, "heartbeat_succeeded"},
+      {heartbeat_failed_event, "heartbeat_failed"},
+      {host_checks_health_changed_event, "host_checks_health_changed"},
+      {host_checks_selected_event, "host_checks_selected"},
+      {software_updates_discovery_requested_event, "software_updates_discovery_requested"}
+    ]
+
+    Enum.each(events, fn {event, _} ->
+      assert :ok == ActivityLogger.log_activity(%{event: event, metadata: %{}})
+    end)
+
+    activity_log = Trento.Repo.all(ActivityLog)
+
+    assert Enum.count(activity_log) == length(events)
+
+    for {event, expected_activity_type} <- events do
+      metadata =
+        event
+        |> Jason.encode!()
+        |> Jason.decode!()
```

Comment:
Why do you encode and decode again?

Reply:
because what gets stored is an arrow map, while the event itself is an atom map
```elixir
%{
  "foo" => "bar"
}
# vs
%{
  foo: "bar"
}
```
Any suggestion on different approaches?

Reply:
Maybe something this should work/help:

```elixir
Map.new(map_with_string_keys, fn {k, v} -> {String.to_exisiting_atom(k), v} end)
```

Reply:
tried, not working tho :sweat_smile: 
```
protocol Enumerable not implemented for %Trento.Hosts.Events.HostRegistered (and any other event struct I guess)
```